src/ds5/ds5-factory.cpp

```diff
@@ -568,21 +562,44 @@ namespace librealsense
             {
                 // verify extrinsic calibration table structure
                 auto table = ds::check_calib<ds::rgb_calibration_table>(raw_data);
-                float3 trans_vector = table->translation_rect;
-                float3x3 rect_rot_mat = table->rotation_matrix_rect;
 
-                // check that data rotation and translation are not zero
-                for (auto i = 0; i < 3; i++)
+                if ( (table->header.version != 0 && table->header.version != 0xffff) && (table->header.table_size >= sizeof(ds::rgb_calibration_table) - sizeof(ds::table_header)))
                 {
-                    if (std::fabs(trans_vector[i]) > std::numeric_limits<float>::epsilon())
+                    float3 trans_vector = table->translation_rect;
+                    float3x3 rect_rot_mat = table->rotation_matrix_rect;
+
+                    // check that translation data is valid
+                    auto found = false;
+                    for (auto i = 0; i < 3; i++)
                     {
-                        for (auto j = 0; j < 3; i++)
+                        if (!std::isfinite(trans_vector[i]))
                         {
-                            if (std::fabs(rect_rot_mat(i, j)) > std::numeric_limits<float>::epsilon())
-                                return true;
+                            LOG_DEBUG("Color stream extrinsic is not valid: \n" << table);
+                            return false;
+                        }
+                        if (std::fabs(trans_vector[i]) > std::numeric_limits<float>::epsilon())
+                            found = true;
+                    }
+
+                    // check that rotation data is valid
+                    for (auto i = 0; i < 3; i++)
+                    {
+                        for (auto j = 0; j < 3; j++)
+                        {
+                            if (!std::isfinite(rect_rot_mat(i, j)))
+                            {
+                                LOG_DEBUG("Color stream extrinsic is not valid: \n" << table);
+                                return false;
+                            }
+
+                            if (std::fabs(rect_rot_mat(i,j)) > std::numeric_limits<float>::epsilon())
+                                found = true;
                         }
                     }
+                    if (found)
+                        return true;
```

Comment:
In case the `found` is set to true after translation VnV, then it will not to recognize that the rotations are all zeros.
To rectify I'd add an additional break condition immediately after translation heuristic, and also reset the bool.